app/src/main/kotlin/com/intive/picover/images/viewmodel/ImagesViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.intive.picover.images.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.intive.picover.images.repository.ImagesRepository
+import com.intive.picover.images.viewmodel.FirebaseStorageState.Error
+import com.intive.picover.images.viewmodel.FirebaseStorageState.Loading
+import com.intive.picover.images.viewmodel.FirebaseStorageState.Success
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class ImagesViewModel @Inject constructor(private val repository: ImagesRepository) : ViewModel() {
+
+	private val _uiState = MutableStateFlow<FirebaseStorageState>(Loading)
+	val uiState: StateFlow<FirebaseStorageState> = _uiState
+
+	init {
```

Comment:
to consider:

I would extend LifecycleViewModel and do the 

viewModelScope.launch {
			getImages()
		}

in onFirstCreate, to be consistent with one approach in the app. Of course it is something we could discuss

Reply:
As we have discussed during the Gild meeting, for now, we will use the init approach.